spec/lib/common/connection-spec.js

```diff
@@ -5,112 +6,255 @@
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            }
+                        });
+                    });
+                    return this.subject.start().then(function () {
+                        return self.subject.connected.should.equal(true);
+                    });
+                });
             });
 
-            it('should be true if connected', function () {
-                var self = this;
+            describe('exchanges', function () {
+                it('should be undefined if not connected', function () {
+                    expect(this.subject.exchanges).equal(undefined);
+                });
 
-                return this.subject.start().then(function () {
-                    return self.subject.connected.should.equal(true);
+                it('should have exchanges if connected', function () {
+                    var self = this;
+
+                    amqp.createConnection = sandbox.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            },
+                            'exchanges':
+                                'on.test'
+                        });
+                    });
+                    return this.subject.start().then(function () {
+                        self.subject.exchanges.should.not.be.undefined;
+                    });
                 });
             });
-        });
 
-        describe('exchanges', function () {
-            it('should be undefined if not connected', function () {
-                expect(this.subject.exchanges).equal(undefined);
-            });
+            describe('exchange', function ()  {
+                it('should not provide an exchange if connection not established ', function () {
+                    return expect(this.subject.exchange()).to.be.rejectedWith(Error);
+                });
 
-            it('should have exchanges if connected', function () {
-                var self = this;
+                it('should provide an exchange if connection established and options is a valid object', function () {
+                    var self = this;
+                    amqp.createConnection = sandbox.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            },
+                            'exchange': function (a, b, callback){
+                                callback('Hello');
+                            },
+                            'exchanges':
+                                'on.test'
+                        });
+                    });
+                    return this.subject.start().then(function () {
+                        if (self.subject.connected === true) {
```

Comment:
Do we want to fail this test if connected !== true? Otherwise I'm worried about the potential for this test producing a false positive as code changes in the future.